### DISCUSSION TITLE
high memory usage

### DISCUSSION THREAD

**Discussion by master-hax** [shizunge/endlessh-go#132] [General]
endlessh-go is currently using 22 megabytes of memory on my system
```
     Memory: 22.1M (peak: 22.6M)
        CPU: 2min 8.315s
```
a week ago it was 15 megabytes, whereas the readme says we should expect about 10 megabytes

it doesn't look like there's a higher volume of connections today compared to last week, so is the memory usage simply from  remembering all the metrics for the past week? is there an easy way to limit this without restarting endlessh-go every week or similar? thanks!

**Comment by shizunge** [comment]
I would say 20MB is still reasonable. The usage is at the same order of magnitude. I now have 3 instances, each use about 14MB, 15MB, 48MB, really depends on the number of connections.

It indeed keeps all the metrics, you can use the following switch to clean inactive ones.

```
  -prometheus_clean_unseen_seconds int
        Remove series if the IP is not seen for the given time. Set to 0 to disable. (default 0)
```

**Comment by master-hax** [comment]
Awesome, let me give this a try.
